Utilities/DavixAdaptor/src/DavixFile.cc

```diff
@@ -0,0 +1,363 @@
+#include "Utilities/DavixAdaptor/interface/DavixFile.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/Utilities/interface/EDMException.h"
+#include "FWCore/Utilities/interface/Exception.h"
+#include <cassert>
+#include <davix.hpp>
+#include <errno.h>
+#include <fcntl.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <vector>
+
+using namespace Davix;
+
+DavixFile::DavixFile(void) {}
+
+DavixFile::DavixFile(const char *name, int flags /* = IOFlags::OpenRead */, int perms /* = 066 */) {
+  open(name, flags, perms);
+}
+
+DavixFile::DavixFile(const std::string &name, int flags /* = IOFlags::OpenRead */,
+                     int perms /* = 066 */) {
+  open(name.c_str(), flags, perms);
+}
+
+DavixFile::~DavixFile(void) {
+  configureDavixLogLevel();
+  close();
+  return;
+}
+
+void DavixFile::close(void) {
+  if (davixPosix && m_fd) {
+    DavixError *err;
```

Comment:
`DavixError* err = nullptr;` since it is unclear what the API will do if you pass a non-null.